src/time_schedule_parser.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'active_support/core_ext/numeric/time'
+require 'tzinfo'
+
+# Parse time definition strings for validity
+class TimeScheduleParser
+  attr_reader :weekdays
+
+  TimeSchedule = Struct.new(:day_from, :day_to, :hour_from, :minute_from, :hour_to, :minute_to, :timezone)
+
+  def initialize
+    @weekdays = %w[MON TUE WED THU FRI SAT SUN]
```

Comment:
Since this never changes, it could be made into a class level constant, e.g. `TimeScheduleParser::WEEKDAYS`, which can be defined earlier in this class.